server/src/main/java/org/opensearch/action/search/SearchRequestOperationsListener.java

```diff
@@ -20,35 +20,35 @@
  * @opensearch.internal
  */
 @InternalApi
-interface SearchRequestOperationsListener {
+abstract class SearchRequestOperationsListener {
```

Comment:
@msfroh @ansjcy @andrross this is a followup on https://github.com/opensearch-project/OpenSearch/pull/11242 to properly encapsulate these APIs as package protected.

The interface -> abstract class change is necessary since the interfaces in Java could be package protected but could not have package protected visibility (limitation), the abstract classes could.